manager/apis/app/v1alpha1/fybrikmodule_types.go

```diff
@@ -144,6 +144,7 @@ type ModuleCapability struct {
 	SupportedInterfaces []ModuleInOut `json:"supportedInterfaces"`
 
 	// API indicates to the application how to access the capabilities provided by the module
+	// TODO This is optinal but in ModuleAPI the endpoint is required?
 	// +optional
 	API *ModuleAPI `json:"api,omitempty"`
```

Comment:
ModuleCapabilities should not be a list. It should be a map where the key is the capability. Otherwise, we just add more and more assumptions, for example: there can not be two read capabilities with different scopes, etc.
If there are two read capabilities with a different scope/api/actions,etc. - we need another FybrikModule definition.
It becomes very complicated to do synchronizations, e.g. select a read module that can read from s3 and redact. You need to make sure that both refer to the same entry, in case there are several entries for read.

Reply:
@shlomitk1 as discussed, the ModuleCapabilities is now changed to a map.

Reply:
@froesef and I consciously decided to make it a list, since you could have multiple instances in the list of the same capability but (for example) with different interfaces.  Please do not make changes to fybrikmodule unless absolutely necessary, since there are other teams already developing using this structure.

Reply:
@simanadler the change was reverted - ModuleCapabilities remains a list